src/components/Dashboard/SpeechRecognition/SpeechRecognition.jsx

```diff
@@ -3,87 +3,96 @@ import {
   OverlayTrigger,
   Tooltip,
   Button,
-  Spinner,
 } from 'react-bootstrap';
-import recordAudio from './recordAudio';
-import sendAudioToWatson from './sendAudioToWatson';
 import microphoneIcon from '../../assets/speakrec.svg';
-
+import SpeechRecognitionHelper from './SpeechRecognitionHelper';
 
 class SpeechRecognition extends Component {
+  recognition = null;
+
   constructor(props) {
     super(props);
+
+    if (!SpeechRecognitionHelper.isSupported()) {
+      this.state = {
+        status: 'disabled',
+      };
+      return;
+    }
+
+    const lang = props.lang || 'en-US';
+    this.recognition = new SpeechRecognitionHelper({
+      continuous: false,
+      interimResults: false,
+      maxAlternatives: 1,
+      lang,
+    }, this.onResult);
     this.state = {
-      recorder: '',
-      status: 'waiting',
+      status: 'ready',
     };
   }
 
-  startRecording = async () => {
+  componentWillUnmount() {
     const { status } = this.state;
-    if (status !== 'waiting') {
+    if (status !== 'recording') {
       return;
     }
-    const recorder = await recordAudio();
-    recorder.start();
 
-    this.setState({
-      status: 'recording',
-      recorder,
-    });
-  };
+    this.recognition.stop();
+  }
 
-  stopRecording = async () => {
-    const { recorder, status } = this.state;
-    const { setText } = this.props;
-    if (status !== 'recording') {
-      return;
+  toggleRecording = async () => {
```

Comment:
can be shortened to avoid switch-case construction, e.g.

```
toggleRecording = () => {
    const { status } = this.state;
    this[status ==='ready' ? 'startRecording' : 'stopRecording']();
  };

  startRecording = () => {
    this.setState({status: ''recording'});
    this.recognition.start();
  };
```

Reply:
Awwww
It looks strange..